### PR TITLE
Bug fix: Fields not honouring schema defined decimal places

### PR DESCRIPTION
**Pull request by Hajara-I**
### What is the context of this PR?
Fix bug that changed number of decimal places after submitting and returning to edit percentage field. 

### How to review 
1. Run `test_percentage` schema, enter a decimal number to three decimal places(dp). Submit then click previous to go back and edit number. Verify that the number hasn't been amended to 2dp and still remains at 3dp. 
2. Run functional test percentage_decimal.spec.js and ensure that test passes.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by Hajara-I** [comment]
[Bug: fields with 1 decimal place automatically populating with 2](https://trello.com/c/kjGnS8HD/5523-bug-fields-with-1-decimal-place-automatically-populating-with-2)

**Review by MebinAbraham — COMMENTED** [review]
Can we make the PR title more meaningful?  Maybe, `Bug fix: Fields not honouring schema defined decimal places`

**Review by rmccar — APPROVED** [review]

File: app/forms/field_handlers/number_handler.py [hunk 1]
```diff
@@ -65,8 +65,17 @@ def _field_type(
...
+                label=self.label,
+                validators=self.validators,
+                description=self.guidance,
+                places=self.max_decimals,
+            )
+            if self._field_type == DecimalFieldWithSeparator
+            else self._field_type(
+                label=self.label, validators=self.validators, description=self.guidance
+            )
         )
```
Comment (MebinAbraham) [inline app/forms/field_handlers/number_handler.py:79]: I recommend only make the additional args conditional. Make it less verbose and fewer refactor needed if things change.

```py
        additional_args = (
            {"places": self.max_decimals}
            if self._field_type == DecimalFieldWithSeparator
            else {}
        )
        return self._field_type(
            label=self.label,
            validators=self.validators,
            description=self.guidance,
            **additional_args,
        )
```

File: tests/functional/spec/percentage_decimal.spec.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import PercentagePage from "../generated_pages/percentage/block.page.js";
+import PercentageDecimalPage from "../generated_pages/percentage/block-decimal.page.js";
+import SubmitPage from "../generated_pages/percentage/submit.page.js";
+
+describe("Percentage", () => {
+  it("Given a percentage decimal answer, when I return to edit the value it does not amend the decimal places", () => {
```
Comment (MebinAbraham) [inline tests/functional/spec/percentage_decimal.spec.js:6]: Percentage isn't important. So I would update the test to Decimal places

`Given an answer allows 3 decimal places, When I enter a value to 3 decimal places and return to edit the value, Then the answer should be displayed with 3 decimal places`


You will also need another one.


`Given an answer allows 3 decimal places, When I enter a value to 1 decimal place and return to edit the value, Then the answer should be displayed with 3 decimal places`

So enter `3.3`, on return it should be `3.300`.

Comment (Hajara-I) [reply tests/functional/spec/percentage_decimal.spec.js]: Test refactored.

File: schemas/test/en/test_percentage.json [hunk 3]
```diff
@@ -59,6 +59,28 @@
...
+                                        "mandatory": false,
+                                        "type": "Percentage",
+                                        "maximum": {
+                                            "value": 100
+                                        },
+                                        "decimal_places": 3
+                                    }
+                                ],
+                                "id": "question-decimal",
+                                "title": "Title",
```
[the remark this answers is not in the document]
Comment (Hajara-I) [reply schemas/test/en/test_percentage.json]: Titles updated

File: app/forms/field_handlers/number_handler.py (hunk 1 above)
Comment (Hajara-I) [reply app/forms/field_handlers/number_handler.py]: `get_field` refactored as suggested.

**Review by ONS-Guilhem-Forey — APPROVED** [review]
LGTM & tested

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**